src/main/java/org/opensearch/knn/index/codec/KNN990Codec/NativeEngines990KnnVectorsReader.java

```diff
@@ -42,11 +44,22 @@ public class NativeEngines990KnnVectorsReader extends KnnVectorsReader {
     private final FlatVectorsReader flatVectorsReader;
     private final SegmentReadState segmentReadState;
     private Map<String, String> quantizationStateCacheKeyPerField;
+    private final Map<String, String> indexToVectorFileName = new HashMap<>();
 
-    public NativeEngines990KnnVectorsReader(final SegmentReadState state, final FlatVectorsReader flatVectorsReader) throws IOException {
+    public NativeEngines990KnnVectorsReader(final SegmentReadState state, final FlatVectorsReader flatVectorsReader) {
         this.segmentReadState = state;
         this.flatVectorsReader = flatVectorsReader;
         loadCacheKeyMap();
+        fillIndexToVectorFileName(state);
+    }
+
+    private void fillIndexToVectorFileName(SegmentReadState state) {
```

Comment:
why not just pass fieldInfos and segmentInfo instead of passing state? Also, can we return Map instead of sharing from class?

Reply:
We can do it, but what for?
This method's job is to fill the table from `SegmentReadState`.
Do you see any side effects?